jenkins/linux_c_option_test.sh

```diff
@@ -59,7 +59,7 @@ declare -a arr=(
     "-Drun_e2e_tests=ON -Duse_baltimore_cert=ON"
     "-Duse_prov_client:BOOL=ON -Dhsm_type_symm_key:BOOL=ON"
     "-Duse_prov_client:BOOL=ON -Dhsm_type_x509:BOOL=ON"
-    "-Duse_prov_client:BOOL=ON -hsm_type_sastoken:BOOL=ON"
+    "-Duse_prov_client:BOOL=ON -Dhsm_type_sastoken:BOOL=ON"
```

Comment:
I'm a bit worried this worked before... I'm still ramping up to CMake but if anyone has an idea on how to actually _test_ this works, please do add your comments.

Reply:
If there is a "treat warnings as errors" in CMake we should enable it. Alternatively, in the future, we can grep the output and check for warnings that shouldn't be there.
For now, we can manually verify if the expected things actually got built whenever we touch this file.

Reply:
it worked because of the way CMake accepts anything that you send it.  by have "" around the below item  it took the whole item as the input which cmake will happily accepts.  Removing the "" will make each variable be used individually.

---
In reply to: [345464418](https://github.com/Azure/azure-iot-sdk-c/pull/1307#discussion_r345464418) [](ancestors = 345464418)